simpleviewinc/keg-herkin
# Zen 599 sidebar scroll

**Ticket**: [ZEN-599](https://jira.simpleviewtools.com/browse/ZEN-599)

## Context

When there are many items added to the side-out menu , they pass beyond the bottom edge of the screen. We are not able to view or interact with items at the bottom since there is no scroll bar.

## Goal

Provide a scrollbar to the side-out menu as well as appropriate bottom padding so that all items can be accessed.

## Updates

- repos/tap/components/sidebar/content/fileTreePanel.js
Added a new view/div that wraps around the menu child items . Added a scrollbar to this element along with reStyle with the necessary height and bottom padding

## Testing

Docker image : https://github.com/simpleviewinc/keg-herkin/pkgs/container/keg-herkin

**Setup**

- Pull this PR => keg herkin && keg pr 42
- Navigate to repos/example folder and then start keg herkin => keg herkin start
- Open launched app at http://herkin-zen-599-sidebarscroll.local.keghub.io/
- Escape from the pop up window so you are on the main site

**Test**

- Open the side bar menu and expand all sub sections 
- Add feature 10-15 feature files by clicking on the plus symbol in the 'test files' header
- The items towards the bottom should have disappeared from view
- hover over the sidebar menu and scroll. 
- you should be able to interact with items in the menu, even items that have moved below the bottom page boundary
